xray/segment_model.go

```diff
@@ -21,7 +21,7 @@ type Segment struct {
 	sync.Mutex
 	parent           *Segment
 	openSegments     int
-	totalSubSegments int
+	totalSubSegments int64
```

Comment:
I am wondering why we have to change the variable type at here?

Reply:
Changed the type so we can use the atomic package directly and make it thread safe.